nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/ConsumeGCPubSub.java

```diff
@@ -0,0 +1,190 @@
+
+@SeeAlso({PublishGCPubSub.class})
+@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@Tags({"google", "google-cloud", "pubsub", "consume"})
+@CapabilityDescription("Consumes message from the configured Google Cloud PubSub subscription. If the 'Batch Size' is set, " +
+        "the configured number of messages will be pulled in a single request, else only one message will be pulled.")
+@WritesAttributes({
+        @WritesAttribute(attribute = ACK_ID_ATTRIBUTE, description = ACK_ID_DESCRIPTION),
+        @WritesAttribute(attribute = SERIALIZED_SIZE_ATTRIBUTE, description = SERIALIZED_SIZE_DESCRIPTION),
+        @WritesAttribute(attribute = MSG_ATTRIBUTES_COUNT_ATTRIBUTE, description = MSG_ATTRIBUTES_COUNT_DESCRIPTION),
+        @WritesAttribute(attribute = MSG_PUBLISH_TIME_ATTRIBUTE, description = MSG_PUBLISH_TIME_DESCRIPTION),
+        @WritesAttribute(attribute = DYNAMIC_ATTRIBUTES_ATTRIBUTE, description = DYNAMIC_ATTRIBUTES_DESCRIPTION)
+})
+public class ConsumeGCPubSub extends AbstractGCPubSubProcessor {
+
+    public static final PropertyDescriptor SUBSCRIPTION = new PropertyDescriptor.Builder()
+            .name("gcp-pubsub-subscription")
+            .displayName("Subscription")
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .description("Subscription name of the Google Cloud Pub/Sub")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    private SubscriberStub subscriber = null;
+    private PullRequest pullRequest;
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        final Integer batchSize = (context.getProperty(BATCH_SIZE).isSet()) ? context.getProperty(BATCH_SIZE).asInteger() : 1;
+
+        pullRequest = PullRequest.newBuilder()
+                .setMaxMessages(batchSize)
+                .setReturnImmediately(false)
+                .setSubscription(getSubscriptionName(context))
+                .build();
+
+        try {
+            subscriber = getSubscriber(context);
+        } catch (IOException e) {
+            getLogger().error("Failed to create Google Cloud Subscriber due to ", e);
+        }
+    }
+
+    @OnStopped
+    public void onStopped() {
+        if (subscriber != null) {
+            subscriber.shutdown();
+        }
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return ImmutableList.of(PROJECT_ID,
+                GCP_CREDENTIALS_PROVIDER_SERVICE,
+                SUBSCRIPTION,
+                BATCH_SIZE);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Collections.singleton(REL_SUCCESS);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        if (subscriber == null) {

```

Comment:
Block L102:106 covers that, right? If `getSubscriber()` fails to create the subscriber, it will throw the exception which is logged. I'm fine with adding a log here but wondering whether that block alone is not enough.

Reply:
Yep, I'm thinking about an edge case: let's say the user see the bulletin generated by the OnScheduled call, but move to something else for some reasons, then the bulletin won't show up anymore after 5 minutes even though the processor is running just fine. Right?

Reply:
Got your point. A log has to be put here. I'll do it.

Reply:
It's done.